check.ts

```diff
@@ -0,0 +1,140 @@
+import { Request, Response } from 'express';
+
+import ReviewService from '@/services/reviewService';
+import type { PullRequestWebhookEvent, PullRequestReviewWebhookEvent, IssueCommentWebhookEvent } from '@/types';
+import config from '@/utils/config';
+import logger from '@/utils/logger';
+
+class WebhookController {
+  private reviewService: ReviewService;
+
+  constructor() {
+    this.reviewService = new ReviewService();
+  }
+
+  async handlePullRequest(req: Request<{}, {}, PullRequestWebhookEvent>, res: Response): Promise<void> {
+    try {
+      const { action, number, pull_request, repository } = req.body;
+
+      logger.info('Received PR webhook', {
+        action,
+        number,
+        repo: repository?.full_name,
+      });
+
+      // Only process certain actions
+      const reviewActions = ['opened', 'synchronize', 'reopened'];
+      if (!reviewActions.includes(action)) {
+        logger.info('Ignoring PR action', { action });
+        return res.status(200).json({ message: 'Action ignored' });
+      }
+
+      // Only process PRs for configured repository
+      // const owner = repository.owner.login;
+      // const repo = repository.name;
+
+      // if (owner !== config.GITHUB_OWNER || repo !== config.GITHUB_REPO) {
+      //   logger.info('Ignoring PR from different repository', {
+      //     owner,
+      //     repo,
+      //     configured: `${config.GITHUB_OWNER}/${config.GITHUB_REPO}`,
+      //   });
+      //   return res.status(200).json({ message: 'Repository not configured for review' });
+      // }
+
+      this.processReviewAsync(repository.owner.login, repository.name, number);
+    } catch (error: any) {
+      logger.error('Error handling PR webhook:', { error: error.message });
+      res.status(500).json({ error: 'Internal server error' });
+    }
+  }
+
+  private async processReviewAsync(owner: string, repo: string, pullNumber: number): Promise<void> {
+    try {
+      logger.info('Starting async PR review', { owner, repo, pullNumber });
+
+      const result = await this.reviewService.reviewPullRequest(owner, repo, pullNumber);
+
+      logger.info('Async PR review completed', {
+        owner,
+        repo,
+        pullNumber,
+        status: result.status,
+      });
+    } catch (error: any) {
+      logger.error('Error in async PR review:', {
+        error: error.message,
+        stack: error.stack,
+        owner,
+        repo,
+        pullNumber,
+      });
+    }
+  }
+
+  async handlePullRequestReview(req: Request<{}, {}, PullRequestReviewWebhookEvent>, res: Response): Promise<void> {
```

Comment:
Missing webhook signature verification. GitHub webhooks should be validated using the `X-Hub-Signature-256` header to ensure the request is actually from GitHub. This is a critical security requirement for webhook endpoints.

```suggestion
// Add signature validation in constructor or as middleware
private validateWebhookSignature(payload: string, signature: string): boolean {
  const expectedSignature = crypto
    .createHmac('sha256', config.GITHUB_WEBHOOK_SECRET)
    .update(payload)
    .digest('hex');
  return crypto.timingSafeEqual(
    Buffer.from(`sha256=${expectedSignature}`),
    Buffer.from(signature)
  );
}
```